queue/poller.go

```diff
@@ -0,0 +1,47 @@
+package queue
+
+import (
+	"database/sql"
+
+	"github.com/lbryio/transcoder/pkg/worker"
+	"github.com/pkg/errors"
+)
+
+type Poller struct {
+	queue               *Queue
+	incomingTasks       chan *Task
+	incomingTaskCounter uint64
+}
+
+func (p *Poller) Process() error {
+	t, err := p.queue.Poll()
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil
+		}
+		return errors.Wrap(worker.FatalError, err.Error())
+	}
+	p.incomingTasks <- t
```

Comment:
if Shutdown() is called right before this, the write to the closed channel will panic